elastic/kibana
# [TIP] Enables TI plugin with kibana.yml feature flag

## Summary

The Threat Intelligence plugin (introduced in [this PR](https://github.com/elastic/kibana/pull/136479)) and is not yet ready for General Availability. This PR hides the new plugin behind a feature flag. As the plugin is dynamically loaded within the Security Solution plugin, we're leveraging the already existing `allowedExperimentalValues` constant to:
- show/hide the navbar entries in both old and new navigation
- load the plugin

### How to test

Threat Intelligence plugin disabled:

- start Kibana without adding any Threat Intelligence entry to your [kibana.yml](https://github.com/elastic/kibana/blob/main/config/kibana.yml)
- check that the in the old navigation, the Threat Intelligence entry doesn't appear in the Explore section of the sidenav
- check that in the new navigation, the Threat Intelligence panel doesn't appear in the Explore section
- check that trying to deep link to http://localhost:5601/app/security/threat_intelligence redirects to http://localhost:5601/app/security/get_started

https://user-images.githubusercontent.com/17276605/182380842-f12e2a49-4128-4531-9218-ae747fd3613a.mov

Threat Intelligence plugin enabled:

- in your [kibana.yml](https://github.com/elastic/kibana/blob/main/config/kibana.yml) file, add the following entry at the bottom `xpack.securitySolution.enableExperimental: ['threatIntelligenceEnabled']`
- check that 

https://user-images.githubusercontent.com/17276605/182380964-72016ab2-3c44-4eec-9c59-2d2d853ca48b.mov

When the Threat Intelligence plugin is ready for General Availability, we should be able to simply do a revert of this commit! :) 

https://github.com/elastic/security-team/issues/4505

### Checklist

Delete any items that are not applicable to this PR.
